tests/test_eslint_flat_config_bug.py

```
@@ -0,0 +1,452 @@
+            config = {
+                "scripts": {"lint": "eslint src/ --ext .ts,.tsx"}
+            }
+            package_json.write_text(json.dumps(config, indent=2))
+
+            linter = ESLintLinter(project_root=str(project_root))
+
+            # Build command will add --format=json which may break
+            command = linter.build_command(["src/index.ts"])
+
+            # Verify the problematic command is generated
+            assert "--format=json" in command
```

Ruff (0.12.2)

Use of `assert` detected

(S101)